fission-core/src/capabilities/fission.rs

```diff
@@ -211,65 +214,166 @@ mod tests {
     #[test_log::test]
     #[ignore]
     fn test_powerbox_ucan_resource() -> TestResult {
-        let store = InMemoryStore::<RawCodec>::default();
-        let did_verifier_map = DidVerifierMap::default();
+        // let store:  = ucan::delegation::store::MemoryStore::default();
+
+        let server_sk = ed25519_dalek::SigningKey::generate(&mut thread_rng());
+        let server_signer =
+            ucan::did::preset::Signer::Key(ucan::did::key::Signer::EdDsa(server_sk));
+        let server_ed_did_key = EdDidKey::new(server_sk);
+
+        let server = ucan::did::preset::Verifier::Key(ucan::did::key::Verifier::EdDsa(
+            server_sk.verifying_key(),
+        ));
 
-        let server = &EdDidKey::generate();
-        let device = &EdDidKey::generate();
+        let device = ucan::did::preset::Verifier::Key(ucan::did::key::Verifier::EdDsa(
+            ed25519_dalek::SigningKey::generate(&mut thread_rng()).verifying_key(),
+        ));
+
+        // FIXME perhaps add this back upstream as a named const
+        let varsig_header = ucan::crypto::varsig::header::EdDsaHeader {
+            codec: libipld::cbor::DagCborCodec,
+        };
+
+
+        // 1.               account -*-> server
+        // 2.                            server -a-> device
+        // 3.  dnslink -d-> account
+        // 4. [dnslink -d-> account -*-> server -a-> device]
 
         // Both of these UCANs just create ephemeral DIDs & delegate all of those
         // DID's rights to the server
-        let (account_did, account_ucan) = server_create_resource(server)?;
-        let (dnslink_did, dnslink_ucan) = server_create_resource(server)?;
+        let (account_did, _account_ucan) = server_create_resource(&server_ed_did_key)?;
+        let (dnslink_did, _dnslink_ucan) = server_create_resource(&server_ed_did_key)?;
+        let mut seed = vec![];
+
+        // 1.               account -*-> server
+        let account_pbox = ucan::delegation::Payload::powerbox(
+            account_did,
+            server,
+            "/".into(),
+            ucan::time::Timestamp::five_years_from_now(),
+        );
```

Comment:
I threw this helper together quickly:

```rust
    pub fn powerbox(issuer: DID, audience: DID, command: String, expiration: Timestamp) -> Self {
        let mut seed = vec![];
        let nonce = Nonce::generate_12(seed.as_mut());

        Payload {
            issuer,
            subject: None,
            audience,
            command,
            policy: vec![],
            metadata: BTreeMap::new(),
            nonce,
            expiration,
            not_before: None,
        }
    }
```

I expect that we'll have a bunch of these kind of utilities

Reply:
I keep thinking about switching `command: String` to something more `struct`ured, but ultimately this just does string comparison at runtime

Reply:
It's an interesting idea to add helpers for these :)

IMO it would still make sense to expose a builder, so e.g. you'd have a nonce set by default, don't need to set `metadata` or `not_before` everytime, etc.

Reply:
Great idea; I'll add that with the builder macro now 👍

Reply:
BOOM
<img width="984" alt="Screenshot 2024-03-07 at 10 27 36" src="https://github.com/fission-codes/fission-server/assets/1052016/840c05a0-04d3-4da2-b749-9d8b1e393d62">

Reply:
Ah sweet. It uses `&mut Self` style. My personal fav

Reply:
I might actually extend this to the `Delegation` envelope, too, so that you don't have to manually wrap: on `build` (maybe a different name?) it will run signing etc. This also avoids having to lift multiple layers of failure (inner builder / outer builder)